src/main/java/uk/gov/companieshouse/disqualifiedofficersdataapi/service/DisqualifiedOfficerService.java

```diff
@@ -56,109 +54,55 @@ public DisqualifiedOfficerService(DisqualifiedOfficerRepository repository,
+            saveAndCallChsKafka(contextId, officerId, document, DisqualificationResourceType.CORPORATE,
+                    existingDocument.orElse(null));
         } else {
-            deleteNaturalDisqualification(contextId, officerId);
+            logger.info(STALE_DELTA_AT_MESSAGE);
         }
     }
 
-    /**
-     * Delete a corporate disqualification record.
-     * @param contextId passed into the call to changed-resource
-     * @param officerId used to find the document to delete
-     */
-    private void deleteCorporateDisqualification(String contextId, String officerId) {
-        CorporateDisqualificationDocument document = retrieveCorporateDisqualification(officerId);
-
-        document.getData().setKind(KindEnum.CORPORATE_DISQUALIFICATION);
-        disqualifiedOfficerApiService.invokeChsKafkaApi(
-                new ResourceChangedRequest(contextId, officerId,
-                        DisqualificationResourceType.CORPORATE, document.getData(), true));
-        logger.info(String.format("ChsKafka api DELETED invoked updated successfully for context id: %s and officer id: %s",
-                contextId,
-                officerId));
-
-        repository.delete(document);
-        logger.info(String.format("Corporate disqualification is deleted in MongoDb for context id: %s and officer id: %s",
-                contextId,
-                officerId));
-    }
-
-    /**
-     * Delete a natural disqualification record.
-     * @param contextId passed into the call to changed-resource
-     * @param officerId used to find the document to delete
-     */
-    private void deleteNaturalDisqualification(String contextId, String officerId) {
-        NaturalDisqualificationDocument document = retrieveNaturalDisqualification(officerId);
-
-        document.getData().setKind(uk.gov.companieshouse.api.disqualification.NaturalDisqualificationApi.
-                KindEnum.NATURAL_DISQUALIFICATION);
-        disqualifiedOfficerApiService.invokeChsKafkaApi(
-                new ResourceChangedRequest(contextId, officerId,
-                        DisqualificationResourceType.NATURAL, document.getData(), true));
-        logger.info(String.format("ChsKafka api DELETED invoked updated successfully for context id: %s and officer id: %s",
-                contextId,
-                officerId));
-
-        repository.delete(document);
-        logger.info(String.format("Natural disqualification is deleted in MongoDb for context id: %s and officer id: %s",
-            contextId,
-            officerId));
-    }
-
-    private boolean isLatestRecord(OffsetDateTime deltaAt, DisqualificationDocument existingDocument) {
-        return StringUtils.isBlank(existingDocument.getDeltaAt()) ||
-                !deltaAt.isBefore(ZonedDateTime.parse(existingDocument.getDeltaAt(), FORMATTER)
-                        .toOffsetDateTime());
+    private boolean isLatestRecord(OffsetDateTime requestDeltaAt, DisqualificationDocument existingDocument) {
```

Comment:
It was that adding a DeltaAtHandler class allowed us to verify it's correct usage during the delete because we can mock it and verify which arguments were passed to it.

On top of that, I didn't want to interfere too much with refactoring the upsert service as that would mean the upsert flow would need smoke testing manually because we don't have regression tests for disqualifications.

I think once we create regression tests in Karate for disqualifications, we can do a more invasive refactor.

What I will do is revert it to what it was so there's no difference from the previous implementation.

Reply:
On second thought, I don't think manually smoke testing the delta_at logic on an upsert is too much extra work. So i will make the changes.